File: limiteMEI-api/src/main/java/com/limiteMEI/limiteMEI/dto/dashboard/DashboardDTO.java
package com.limiteMEI.limiteMEI.dto.dashboard;

import com.limiteMEI.limiteMEI.dto.mei.ApuracaoMeiDTO;
import com.limiteMEI.limiteMEI.dto.movimento.MovimentoFinanceiroDTO;
import lombok.*;
import java.math.BigDecimal;
import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DashboardDTO {
    private String empresa;
    private Integer ano;
    private Integer mes;
    private BigDecimal saldoTotal;
    private BigDecimal entradasMes;
    private BigDecimal saidasMes;
    private BigDecimal contasReceber;
    private BigDecimal contasPagar;
    private BigDecimal vencidoReceber;
    private BigDecimal vencidoPagar;
    private Integer quantidadeVencidos;
    private List<DashboardContaDTO> contas;
    private List<MovimentoFinanceiroDTO> ultimasMovimentacoes;
    private ApuracaoMeiDTO mei;
}

File: limiteMEI-angular/src/app/dtos/dashboard/dashboard.dto.ts
import { MovimentoFinanceiroDTO } from '../movimento/movimento.financeiro';
import { ApuracaoMeiDTO } from '../mei/apuracao-mei.dto';
import { DashboardObrigacaoMeiDTO } from '../mei/obrigacao-mei.dto';

export interface DashboardContaDTO {
  id: number;
  nome: string;
  saldo: number;
}

export interface DashboardDTO {
  empresa: string;
  ano: number;
  mes: number;
  saldoTotal: number;
  entradasMes: number;
  saidasMes: number;
  contasReceber: number;
  contasPagar: number;
  vencidoReceber: number;
  vencidoPagar: number;
  quantidadeVencidos: number;
  contas: DashboardContaDTO[];
  ultimasMovimentacoes: MovimentoFinanceiroDTO[];
  mei: ApuracaoMeiDTO;
  obrigacaoMei: DashboardObrigacaoMeiDTO;
}
